noir-projects/noir-contracts/contracts/account/ecdsa_k_account_contract/src/main.nr

```diff
@@ -27,16 +27,40 @@ pub contract EcdsaKAccount {
         let this = context.this_address();
         let pub_key_note = EcdsaPublicKeyNote::new(signing_pub_key_x, signing_pub_key_y, this);
 
-        storage.public_key.initialize(pub_key_note).emit(encode_and_encrypt_note(
+        // Safety: The sender for tags is only used to compute unconstrained shared secrets for emitting logs.
+        // Since this value is only used for unconstrained tagging and not for any constrained logic,
+        // it is safe to load from an unconstrained context.
+        // We use `unwrap_or` here because we need the value to be defined in the call to `set_sender_for_tags`
+        // at the end of this function.
+        // TODO: The "unwrap_or" below is being run in a constrained context but it doesn't need to be constrained.
+        // Should we optimize this?
+        let original_sender = unsafe { get_sender_for_tags().unwrap_or(this) };
```

Comment:
The naming here is incorrect because if the sender for tags in undefined we default to this. So in the `undefined` case it's a lie.

I think the correct way to handle this is to update `set_sender_for_tags` to accept an option on the input and hence allow for resetting sender to undefined.

Would the reviewer agree or am I overthinking this?